src/CoreWCF.Http/src/CoreWCF/WSHTTPSecurity.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text;
+using CoreWCF.Channels;
+using CoreWCF.Runtime;
+using CoreWCF.Security;
+using CoreWCF;
+
+namespace CoreWCF
+{
+    public sealed class WSHttpSecurity
+    {
+        internal const SecurityMode DefaultMode = SecurityMode.Message;
+
+        SecurityMode mode;
+        HttpTransportSecurity transportSecurity;
+        NonDualMessageSecurityOverHttp messageSecurity;
+
+        public WSHttpSecurity()
+            : this(DefaultMode, GetDefaultHttpTransportSecurity(), new NonDualMessageSecurityOverHttp())
+        {
+        }
+
+        internal WSHttpSecurity(SecurityMode mode, HttpTransportSecurity transportSecurity, NonDualMessageSecurityOverHttp messageSecurity)
+        {
+            this.mode = mode;
+            this.transportSecurity = transportSecurity == null ? GetDefaultHttpTransportSecurity() : transportSecurity;
+            this.messageSecurity = messageSecurity == null ? new NonDualMessageSecurityOverHttp() : messageSecurity;
+        }
+
+        internal static HttpTransportSecurity GetDefaultHttpTransportSecurity()
+        {
+            HttpTransportSecurity transportSecurity = new HttpTransportSecurity();
+            transportSecurity.ClientCredentialType = HttpClientCredentialType.Windows;
+            return transportSecurity;
+        }
+
+        public SecurityMode Mode
+        {
+            get { return this.mode; }
+            set
+            {
+                if (!SecurityModeHelper.IsDefined(value))
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException("value"));
+                }
```

Comment:
Use nameof(value) instead. This is likely going to be throughout the codebase a lot so a find/replace might be worth using.